subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/InstantExecutionReport.kt

```diff
@@ -119,45 +162,52 @@ class InstantExecutionReport(
 
     private
     fun errors() =
-        problems.filterIsInstance<PropertyProblem.Error>()
+        problems.toList().filterIsInstance<PropertyProblem.Error>()
```

Comment:
⭕️ `problems.asIterable()` instead?

Reply:
`.toList()` is used here to take a copy of the list before filtering because it can be concurrently modified. IIUC `.asIterable()` won't fix that.

Reply:
From the docs:
> the iterator is guaranteed not to throw ConcurrentModificationException

Reply:
Yep, but the kotlin implementation of `Iterable.asIterable()` is `return this` ...

Reply:
Ah but then `filterInstance()` should use the iterator which should be safe. Correct?

Reply:
That's my understanding as well.

Reply:
Done